app/controllers/masterpieces_controller.rb

```diff
@@ -15,6 +15,8 @@ def show
   # GET /masterpieces/new
   def new
     @masterpiece = Masterpiece.new
+    @recipe_id = masterpiece_params[:recipe_id]
+    @user_id = current_user.id if current_user
   end
 
```

Comment:
In masterpiece controller, you are looking for masterpiece_params[:recipe_id] when you call "new" method.
You are trying to call the "new" method inside recipe's show view. But the original version also call "new" method in masterpiece's index view and there will be missing params. It doesn't make sense to create a new masterpiece from the index view directly. Shall we remove the link in masterpiece's index view? Later on we would need to get rid of the masterpiece's index view but for now let's keep the view.